### ISSUE TITLE
Secondary unique indexes lead to divergent conflicts even if violated by the same tuples

### ISSUE THREAD

**Issue by ringerc** [2ndQuadrant/bdr#95]
BDR bails out of apply if it finds a tuple that violates two or more unique indexes (possibly including the `PRIMARY KEY`). It currently fails with:

```
ERROR: diverging uniqueness conflict
```

but 0.10.0 will fail with

```
ERROR: multiple unique constraints violated by remotely INSERTed tuple
```

This is the case even where the same existing tuple conflicts with the incoming tuple on all conflicting indexes. In this case we could still safely perform a last-update-wins resolution, as there are multiple conflicting indexes but only the one pair of conflicting tuples.

This means that it's currently unsafe to use secondary unique indexes in BDR. Any secondary unique index that isn't more specific than the `PRIMARY KEY` (and thus useless, because any tuple that violates it would also violate the `PRIMARY KEY`) can potentially lead to divergent conflicts that block replication.

Additionally, there's no attempt to detect this condition for `UPDATE` at all.

**Comment by anarazel** [comment]
On 2015-06-22 00:40:17 -0700, Craig Ringer wrote:
> This means that it's currently unsafe to use secondary unique indexes in BDR. Any secondary unique index that isn't more specific than the `PRIMARY KEY` (and thus useless, because any tuple that violates it would also violate the `PRIMARY KEY`) can potentially lead to divergent conflicts that block replication.

I don't think that's a accurate description. It's quite often the case
that you'll never have a conflict at all on the primary key because it's
a sequence.

**Comment by ringerc** [comment]
Good point - so *one* secondary unique index is safe, iff the `PRIMARY KEY` originates from a global sequence, uuid, or other globally unique surrogate key generation.
